source/kra/layer.d

```diff
@@ -228,36 +199,41 @@ public:
         int[2] location;
     }
 
-    /**
-        The type of layer
-    */
-    LayerType type;
-
-    /**
-        Blending mode
-    */
-    BlendingMode blendModeKey;
-
-    /**
-        Opacity of the layer
-    */
-    int opacity;
-
     /**
         Whether the layer is visible or not
     */
     bool isVisible;
 
     /**
-        Color mode of layer
+        Sub-layers
     */
-    ColorMode colorMode;
+    Layer[] children;
+    
+    /** 
+        Constructor using tuples of attributes
+    */
+    this(T...)(in T attributes) {
+		name = getAttrValue!string(attributes, "name", "");
```

Comment:
This seems to be using tabs for indentation, please use spaces.
In VSCode the settings would be `4 Spaces`; there's a command called `Convert Indentation to Spaces`; I recommend using this.